### PR TITLE
Move buttons to top right for custom scroll element

### PR DESCRIPTION
**Pull request by dkanada**
The only huge issue is that I haven't figured out how to hide the scroll buttons if the elements don't actually go past the page width. There might also be some style kinks to work out.

This fixes the overlap issue several people mentioned since the buttons no longer show up in the scroller at all. The scroll buttons also no longer break and let you navigate past the last element either.

@dinki can you test these changes with the TV guide element I mentioned a bit ago and let me know if this breaks anything?

### CODE REVIEW

**Comment by dkanada** [comment]
Oh yeah, this fixes the overlap issue several people mentioned since the buttons no longer show up in the scroller at all. The scroll buttons also no longer break and let you navigate past the last element either.

**Comment by dinki** [comment]
I'm not sure when I'll have time to get to this, but you should be able to use the 'fake live tv' files I made to test it. You can also replicate the single channel in the M3U file multiple times, give the copies different names, and assign all the same guide data. That should populate your guide enough to test. I can certainly test but it might not be until mid next week.

**Comment by dinki** [comment]
[livetvtestingfiles.zip](https://github.com/jellyfin/jellyfin-web/files/3166552/livetvtestingfiles.zip)

File: src/components/emby-scrollbuttons/emby-scrollbuttons.css [hunk 1]
```diff
@@ -1,60 +1,20 @@
...
-.emby-scrollbuttons-scrollbutton:after {
-    content: '';
-    display: none !important;
-}
-
-.emby-scrollbuttons-scrollbutton:focus {
-    color: inherit !important;
-}
+    overflow: hidden;
+}
```
Comment (DrPandemic) [inline src/components/emby-scrollbuttons/emby-scrollbuttons.css]: No biggie, but you could set your editor to fix that.

**Comment by DrPandemic** [comment]
This should fix the first element of https://github.com/jellyfin/jellyfin-web/issues/302

**Comment by dkanada** [comment]
The thing that bothers you is what I meant above with hiding the scroll buttons, I have no clue how to easily fix that yet. This is the one of several UI changes I wanted to pull over from Plex with the goal of making Jellyfin more polished.

File: src/components/emby-scrollbuttons/emby-scrollbuttons.css (hunk 1 above)
Comment (dkanada) [reply src/components/emby-scrollbuttons/emby-scrollbuttons.css]: I added an editor config so all files should get standardized eventually.

**Comment by dkanada** [comment]
That file is a dependency though, so we might want to remove it and use the upstream version if we can. What exactly was the issue your change fixed?

Also, what browser are you using? At the moment, only chromium browsers properly hide the buttons when the scroller is too short. Each scroller also needs the afterRefresh method linked as seen [here](https://github.com/jellyfin/jellyfin-web/pull/315/files#diff-182d1e7d3570466ecc775a05b555d587R469) to render properly on load.

**Comment by DrPandemic** [comment]
> so we might want to remove it and use the upstream version if we can

Not sure to understand what this means.

Ohh I was testing on firefox. I just tried on chrome and it totally break it. Oh that's interesting, with my "fix" FF hides arrows when they are not needed, but not on chrome. But without it, it's the inverse.

So, what I'm saying is that the scroller is not hidden on FF with your current PR with the list is too short.

**Comment by oddstr13** [comment]
Reddit post mentioning the arrow problem https://www.reddit.com/r/jellyfin/comments/c2tp5m/very_tiny_bug/

**Comment by dkanada** [comment]
There are still a few underlying problems but this is the easiest solution I found that didn't require sweeping changes to the code.

**Comment by dkanada** [comment]
@DrPandemic without the latest commit the negative margin was broken in the android app and I have no clue why. It added a horizontal scrollbar for no reason.

**Pull request merged**